source_code.py

```
@@ -110,43 +85,65 @@ def query_a(results_queue):
-            smtp_server.quit()
+            try:
+                server = smtplib.SMTP(host, 587, timeout=2)
+                server.starttls()
+            except (socket.gaierror, smtplib.SMTPConnectError, OSError):
+                try:
+                    server = smtplib.SMTP(host, 25, timeout=2)
+                except (socket.gaierror, smtplib.SMTPConnectError, OSError):
+                    try:
+                        server = smtplib.SMTP_SSL(host, 465, timeout=2)
+                    except Exception:
+                        continue
```

Ruff (0.12.2)

`try`-`except`-`continue` detected, consider logging the exception

(S112)